lib/LaTeXML/Post/SVG.pm

```diff
@@ -35,55 +37,62 @@ sub toProcess {
 sub process {
   my ($self, $doc, @svg) = @_;
   local $::IDCOUNTER = 0;
+  if (!$$self{model}) {
+    $$self{model} = LaTeXML::Common::Model->new();
+    $$self{model}->setRelaxNGSchema("LaTeXML");
+    $$self{model}->loadSchema(); }
   $doc->addNamespace($svgURI, 'svg');
-  map { ProcessSVG($_) } @svg;
+  map { ProcessSVG($self, $_) } @svg;
   $doc->adjust_latexml_doctype('SVG');    # Add SVG if LaTeXML dtd.
   return $doc; }
 
 sub getQName {
   my ($node) = @_;
   return $LaTeXML::Post::DOCUMENT->getQName($node); }
 
-sub copy_some_attributes {
-  my ($to, $from, @attributes) = @_;
-  foreach my $key (@attributes) {
-    my $value = $from->getAttribute($key);
-    if (defined $value) {
+sub copy_valid_attributes {
+  my ($self, $to, $from) = @_;
+  my $qname = getQName($to);
+  foreach my $attr ($from->attributes) {
+    my $key = $attr->getName;
+    if ($$self{model}->canHaveAttribute($qname, $key)) {
       $from->removeAttribute($key) if $key eq 'xml:id';
-      $to->setAttribute($key, $value); } }
+      $to->setAttribute($key, $from->getAttribute($key)); } }
   return; }
 
-sub copy_attributes_except {
-  my ($to, $from, @attributes) = @_;
-  my %excluded = map { ($_ => 1) } @attributes;
-  foreach my $attr ($from->attributes) {
-    my $key = $attr->getName;
-    next                         if $excluded{$key};
-    $from->removeAttribute($key) if $key eq 'xml:id';
-    $to->setAttribute($key, $from->getAttribute($key)); }
+sub copy_position {
+  my ($self, $to, $from) = @_;
+  my $x = $from->getAttribute('x');
+  my $y = $from->getAttribute('y');
+  $to->setAttribute(cx => $x) if defined $x;
+  $to->setAttribute(cy => $y) if defined $y;
   return; }
 
+sub to_px {
+  my ($pt) = @_;
+  return ($pt =~ s/pt$// ? $pt * $LaTeXML::Util::Image::DOTS_PER_POINT : $pt); }
+
 ####################################
 ## fixes an svg node
 ####################################
 
 sub ProcessSVG {
-  my ($node) = @_;
+  my ($self, $node) = @_;
```

Comment:
if it takes `$self` as an argument it may be more idiomatic to call as `$self->processSVG` ... 
Same comment for `convertNode`

Reply:
sure.